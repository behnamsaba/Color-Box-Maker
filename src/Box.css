.Box {
    padding: 8px 12px;
    margin:8px 12px;
    position: relative;



    
}

.Box-btn{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}