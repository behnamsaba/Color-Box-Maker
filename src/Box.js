import './Box.css';

const Box = ({ color, width, height,id, removeSquare}) => {
    const divStyle = {
        backgroundColor: color,
        width,
        height
    }


    return(
        <div className="Box" style={divStyle} id={id}>
            <button className="Box-btn" onClick={() => removeSquare(id)}>X</button>
        </div>
    )
}

export default Box;