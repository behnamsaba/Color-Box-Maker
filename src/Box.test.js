import { render } from '@testing-library/react';
import Box from './Box';

test('renders without crashing', () => {
    render(<Box />);
});

test("match snapshot", function() {
  const {asFragment} = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
})