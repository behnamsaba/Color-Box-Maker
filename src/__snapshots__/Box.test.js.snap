// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
